From 657caf5140e1e1133611778425aa34352d093d76 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jose.dapena@lge.com>
Date: Fri, 25 Oct 2019 15:07:09 +0000
Subject: [PATCH 11/62] IWYU: include algorithm to use std::lower_bound in
 ui/gfx/font.cc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[openSUSE chromium-79-include.patch]

Fix GCC build because of missing include:
../../ui/gfx/font.cc: In function ‘gfx::Font::Weight gfx::FontWeightFromInt(int)’:
../../ui/gfx/font.cc:114:8: error: no matching function for call to ‘lower_bound(const gfx::Font::Weight*, const gfx::Font::Weight*, int&, gfx::FontWeightFromInt(int)::<lambda(const gfx::Font::Weight&, const int&)>)’
       });
        ^
In file included from /usr/include/c++/8/bits/char_traits.h:39,
                 from /usr/include/c++/8/string:40,
                 from ../../ui/gfx/font.h:8,
                 from ../../ui/gfx/font.cc:5:
/usr/include/c++/8/bits/stl_algobase.h:984:5: note: candidate: ‘template<class _ForwardIterator, class _Tp> _ForwardIterator std::lower_bound(_ForwardIterator, _ForwardIterator, const _Tp&)’
     lower_bound(_ForwardIterator __first, _ForwardIterator __last,
     ^~~~~~~~~~~
/usr/include/c++/8/bits/stl_algobase.h:984:5: note:   template argument deduction/substitution failed:
../../ui/gfx/font.cc:114:8: note:   candidate expects 3 arguments, 4 provided
       });
        ^

Bug: 819294
Change-Id: Ic59dcf3a06bdd54d1d426c08a61624873a0ff30c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1879909
Commit-Queue: Alexei Svitkine <asvitkine@chromium.org>
Reviewed-by: Alexei Svitkine <asvitkine@chromium.org>
Cr-Commit-Position: refs/heads/master@{#709472}
---
 base/profiler/register_context.h                    | 1 +
 chrome/browser/apps/launch_service/launch_manager.h | 1 +
 ui/gfx/font.cc                                      | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/base/profiler/register_context.h b/base/profiler/register_context.h
index 556b34c27aa6f..4a17e5d13c24f 100644
--- a/base/profiler/register_context.h
+++ b/base/profiler/register_context.h
@@ -9,6 +9,7 @@
 #ifndef BASE_PROFILER_REGISTER_CONTEXT_H_
 #define BASE_PROFILER_REGISTER_CONTEXT_H_
 
+#include <cstdint>
 #include <type_traits>
 
 #include "build/build_config.h"
diff --git a/chrome/browser/apps/launch_service/launch_manager.h b/chrome/browser/apps/launch_service/launch_manager.h
index 00aeb9d9c2a56..76570ead0f9d6 100644
--- a/chrome/browser/apps/launch_service/launch_manager.h
+++ b/chrome/browser/apps/launch_service/launch_manager.h
@@ -6,6 +6,7 @@
 #define CHROME_BROWSER_APPS_LAUNCH_SERVICE_LAUNCH_MANAGER_H_
 
 #include <string>
+#include <vector>
 
 #include "base/macros.h"
 
diff --git a/ui/gfx/font.cc b/ui/gfx/font.cc
index 21367fd729761..92b159e13d10c 100644
--- a/ui/gfx/font.cc
+++ b/ui/gfx/font.cc
@@ -4,6 +4,8 @@
 
 #include "ui/gfx/font.h"
 
+#include <algorithm>
+
 #include "base/strings/utf_string_conversions.h"
 #include "build/build_config.h"
 #include "ui/gfx/platform_font.h"
-- 
2.25.0

